packages/docregistry/src/context.ts

```diff
@@ -262,6 +262,9 @@ class Context<T extends DocumentRegistry.IModel> implements DocumentRegistry.ICo
         return this._populate();
       }
     }).catch(err => {
+      if (err === 'Cancel') {
+        return;
```

Comment:
That's right. The main thing I forgot to ask was, should this `return Promise.reject('Cancel');` or something of the sort as well?

Reply:
I think I would consider the error as "Having been handled" at this point. If the user selects `CANCEL`, and we intend that to be a valid user action, should we be throwing an error?

Reply:
Well, I guess what I'm wondering is whether there's some path where this gets called from an extension that wants to save something and needs to know whether that thing was saved or not. It seems as though that's possible, but if a user hits cancel the extension author never learns that the file did not save. I'm not sure.

Reply:
Oh, that's a good point. Maybe the `SaveHandler` should then catch that error.

Reply:
For reasons I don't understand, the error handler in `SaveHandler` is not being called. Do you see anything obvious @afshin ?